lib/chef/provisioning/machine_spec.rb

```diff
@@ -39,43 +39,26 @@ def name
     #
     # chef-provisioning will do its darnedest to not lose this information.
     #
-    def location
-      chef_provisioning_attr('location')
+    def reference
+      attrs['reference'] || attrs['location']
```

Comment:
Would a falsey (i.e. `nil`, `false`) value for `attrs['reference']` lead to issues here? If so another way could be using fetch: `attrs.fetch('reference') { attrs['location'] }` (lazy block version used)